### ISSUE TITLE
Don't preemptively backup the entire game folder when managing a new game

### ISSUE THREAD

**Issue by Al12rs** [Nexus-Mods/NexusMods.App#899]
Backing up the entire game folder is a slow operation that takes up a lot of diskspace for potentially little benefit for the end users.

While this should be still available as an option in the future, I don't believe this is necessary when starting to manage the game for the first time.

Instead, the app could strategically only backup game files that it actually needs to replace, which actually are unlikely to be many, saving in GBs of diskspace and initial setup time.

The game folder should still be indexed, but files not backed up. 
If game files change, it should be assumed to be from game updating, so files should be indexed but not backed up by default. 

This behaviour should be customizable in the future by bot the Game extension developer and the user.
The game extension developer should be able to define file patterns to backup and file patterns to exclude from backup.

**Comment by Al12rs** [comment]
This should make both development faster and improve the experience for early adopters, skipping long wait times when managing games for the first time.

**Comment by Sewer56** [comment]
I have some comments to make, there's edge cases, but I'm too swamped making the docs accessible right now.

**Comment by Al12rs** [comment]
Yeah this is meant to be a proposal to discuss, happy to do that today before or after Retro.

**Comment by Continous** [comment]
It may be useful too to have an option that doesn't back up the game files, but instead advises the users that there's a hash mismatch between their "clean" files and the file state when they first began managing the game. This could be useful if, for example, the user decides to back up the files on some form of external storage that's not readily accessible like a SD card or network storage.
